### PR TITLE
CB-32: Make search requests using program enrollment component

### PR DESCRIPTION
**Pull request by andela-aolaniran**
# JIRA TICKET NAME:
[CB-32](https://issues.openmrs.org/browse/CB-32): Make search requests using program enrollment component

## SUMMARY:
Make search requests based on programs, patient birthdate range, program enrollment date, program completion date and program duration using program enrollment component

### CODE REVIEW

File: app/js/components/tabs/tabcomponents/programmeComponent.jsx [hunk 1]
```diff
@@ -22,63 +38,233 @@ class ProgrammeComponent extends Component {
...
          * try to populate states on the local
         */
+        // console.log('work flows - ', workflows)
         
     }
     render() {
         let programs = this.state.programs.map((program) => {
             return (
-                <option key={program.uuid} value={program.uuid}>
+                <option key={program.uuid} value={program.name}>
```
Comment (dkayiwa) [inline app/js/components/tabs/tabcomponents/programmeComponent.jsx]: Shouldn't the value be uuid?

File: app/js/components/tabs/tabcomponents/programmeComponent.jsx [hunk 2]
```diff
@@ -22,63 +38,233 @@ class ProgrammeComponent extends Component {
...
                     {program.name}
                 </option>
             );
         });
         let workflows = this.state.workflows.map((workflow) => {
             return (
-                <option key={workflow.uuid} value={workflow.uuid}>
+                <option key={workflow.uuid} value={workflow.name}>
```
Comment (dkayiwa) [inline app/js/components/tabs/tabcomponents/programmeComponent.jsx]: Shouldn't the value be uuid?

File: app/js/components/tabs/tabcomponents/programmeComponent.jsx [hunk 3]
```diff
@@ -22,63 +38,233 @@ class ProgrammeComponent extends Component {
...
             );  
-        })
+        });
         // States will be loaded from this.state.states when populated from backend
         let states = "<option> </option>";
 
         let locations = this.state.locations.map((location) => {
             return (
-                <option key={location.uuid} value={location.uuid}>
+                <option key={location.uuid} value={location.display}>
```
Comment (dkayiwa) [inline app/js/components/tabs/tabcomponents/programmeComponent.jsx]: Shouldn't the value be uuid?

File: app/js/components/tabs/tabcomponents/programmeComponent.jsx [hunk 4]
```diff
@@ -176,8 +362,8 @@ class ProgrammeComponent extends Component {
                 <div className="form-group">
                     <label htmlFor="gender" className="col-sm-2 control-label">Patients belonging to?:</label>
                     <div className="col-sm-6">
-                        <select className="form-control" id="" name="">
-                            <option value="all">All</option>
+                        <select className="form-control" id="location">
+                            <option value="">Select Location ?</option>
```
Comment (dkayiwa) [inline app/js/components/tabs/tabcomponents/programmeComponent.jsx]: What is the question mark used for?

File: app/js/components/tabs/tabcomponents/programmeComponent.jsx (hunk 1 above)
Comment (andela-aolaniran) [reply app/js/components/tabs/tabcomponents/programmeComponent.jsx]: Yes, you are right. I was using the value to get the descriptive name for the label and also to make a search request.
I will refactor this to make the search request by the uuid, and use the descriptive name for the label.

File: app/js/components/tabs/tabcomponents/programmeComponent.jsx (hunk 2 above)
Comment (andela-aolaniran) [reply app/js/components/tabs/tabcomponents/programmeComponent.jsx]: Noted.

File: app/js/components/tabs/tabcomponents/programmeComponent.jsx (hunk 3 above)
Comment (andela-aolaniran) [reply app/js/components/tabs/tabcomponents/programmeComponent.jsx]: Noted.

File: app/js/components/tabs/tabcomponents/programmeComponent.jsx (hunk 4 above)
Comment (andela-aolaniran) [reply app/js/components/tabs/tabcomponents/programmeComponent.jsx]: Yea, I would take off the question mark :)

File: app/js/components/tabs/tabcomponents/programmeComponent.jsx [hunk 5]
```diff
@@ -22,30 +38,202 @@ class ProgrammeComponent extends Component {
...
+            this.setState({ workflows: [], workflow: '' });
+        }
     }
     // Get the states from the workflow.
     getStates(workflows) {
         /** Update state based on the data retreived from the workflows
          * @TODO: State data does not exist on the local, but exists on refapp,
          * try to populate states on the local
         */
+        // console.log('work flows - ', workflows)
```
Comment (dkayiwa) [inline app/js/components/tabs/tabcomponents/programmeComponent.jsx]: Do you need the above line?

Comment (andela-aolaniran) [reply app/js/components/tabs/tabcomponents/programmeComponent.jsx]: I missed that. I don't need that line and will remove it now.

**Pull request merged**